dataclients/kubernetes/kube.go

```diff
@@ -1356,3 +1383,50 @@ func (c *Client) Close() {
 		close(c.quit)
 	}
 }
+
+func (c *Client) fetchDefaultFilterConfigs() map[resourceId]string {
+	if c.defaultFiltersDir == "" {
+		log.Debug("default filters are disabled")
+		return make(map[resourceId]string)
+	}
+
+	filters, err := c.getDefaultFilterConfigurations()
+
+	if err != nil {
+		log.WithError(err).Error("could not fetch default filter configurations")
+		return make(map[resourceId]string)
+	}
+
+	log.WithField("#configs", len(filters)).Debug("default filter configurations loaded")
+
+	return filters
+}
+
+const maxFileSize = 1024 * 1024 // 1MB
+
+func (c *Client) getDefaultFilterConfigurations() (map[resourceId]string, error) {
+	files, err := ioutil.ReadDir(c.defaultFiltersDir)
+	if err != nil {
+		return nil, err
+	}
+
+	filters := make(map[resourceId]string)
+	for _, f := range files {
+		r := strings.Split(f.Name(), ".") // format: {service}.{namespace}
+		if len(r) != 2 || f.IsDir() || f.Size() > maxFileSize {
```

Comment:
f could be also a fifo or a socket, which both would block.
I think you should check for file or symlink (symlink could also target a socket, which is bad, but maybe ok enough)

Reply:
Good point!